src/state/config.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""gateway-api-integrator configuration."""
+
+import dataclasses
+import itertools
+import typing
+
+import ops
+from pydantic import BaseModel, Field, ValidationError
+
+
+class InvalidCharmConfigError(Exception):
+    """Exception raised when a charm configuration is found to be invalid.
+
+    Attrs:
+        msg (str): Explanation of the error.
+    """
+
+    def __init__(self, msg: str):
+        """Initialize a new instance of the InvalidCharmConfigError exception.
+
+        Args:
+            msg (str): Explanation of the error.
+        """
+        self.msg = msg
+
+
+@dataclasses.dataclass(frozen=True)
+class CharmConfig(BaseModel):
+    """A component of charm state that contains the charm's configuration.
+
+    Attrs:
+        gateway_class (_type_): _description_
```

Comment:
Is the `_type_` and `_description_` here a placeholder?

Reply:
Yes, I updated the relevant fields with proper description.